security-demo/README.md

```diff
@@ -0,0 +1,263 @@
+
+
+### Spiffe helper configuration (IMPORTANT)
+
+Spiffe helper is used to execute ghostunnel for creating communciation
+channel over TLS for cloud , edge and user workloads . Please refer to
+the specific configurations in the folders section.
+
+All the helper configurations need to updated with IP addresses and
+ports present in
+
+1\. &lt;SPIRE\_PATH&gt;/helper.conf
+
+2\. &lt;SPIRE\_PATH&gt;/event-bus/event-bus-helper.conf
+
+3\. &lt;SPIRE\_PATH&gt;/user-app/user-app-helper.conf
+
+In the list of To-Dos , there is an item to generate these
+configurations automatically based on environment configuration to avoid
+errors.
+
+Following is a sample configuration for kubeedge event-bus interface:
+
+agentAddress = "/tmp/app-agent.sock"
+
+cmdArgs = "**server** --listen 192.168.56.102:18884 --target
+127.0.0.1:2884 --cacert /opt/spire/event-bus/certs/svid\_bundle.pem
+--keystore /opt/spire/event-bus/certs/new\_bundle.p12 **--allow-uri-san
+spiffe://example.org/downstream-app-event-bus**"
+
+cmd = "/opt/spire/ghostunnel"
+
+certDir = "/opt/spire/event-bus/certs"
+
+renewSignal = "SIGUSR1"
+
+svidFileName = "svid.pem"
+
+svidKeyFileName = "svid\_key.pem"
+
+svidBundleFileName = "svid\_bundle.pem"
+
+### Script Configurations
+
+Environment variable configurations : &lt;SPIRE\_PATH&gt;/edge.env
+
+export CLOUD\_VM\_USER=vm1
+
+export CLOUD\_VM\_PASS=vm1
+
+export CLOUD\_VM\_IP=192.168.56.101
+
+export SPIRE\_PATH=/opt/spire
+
+## How to setup and use
+
+1\. Copy the release files to cloud node and edge node at path /opt/spire/.
+
+2\. Update the IP , port and spire path in the above listed configurations.
+
+3\. In cloud node , execute deploy-cloud.sh.
+
+4\. In edge node , execute deploy-edge.sh.
+
+5\. Register an example device with cloud using following command. Please
+note, in the current version, cloud test application opens 30000 port
+for metadata creation (create pod or device) and 20000 port for
+communication with kubeedge edgehub. In cloud node, execute
+
+    curl -XGET http://127.0.0.1:30000/device -H 'content-type:application/json' -d@/opt/spire/app-binaries/test-device.yaml
+
+6\. Run the light\_mapper application from app\_binaries in edge node. Light mapper
+application is a binary built from
+[*https://github.com/kubeedge/examples/tree/master/led-raspberrypi*](https://github.com/kubeedge/examples/tree/master/led-raspberrypi).
+Usage of the application can be referred in the same page.
+
+For spire server cli usage , please refer to
+[*https://github.com/spiffe/spire*](https://github.com/spiffe/spire).
+
+## Present support
+
+1\. Initial infrastructure deployment.
+
+2\. 1-1 cloud and edge node. Multiple edge nodes are not supported in
+scripts.
+
+3\. Communication from user application and edge application using
+certificates issued by cloud spire server and edge spire server.
+
+## ToDo
+
+1\. Optimization for redundancies in configuration and scripts.
+
+2\. Automate build/release and remove binaries from the package.
```

Comment:
So there is still binaries in the package?

Reply:
No, updated the todo list.